apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.BenchmarkingTool/Program.cs

```diff
@@ -0,0 +1,208 @@
+            RunReadTest(configuration, client, logger, "Read[2]");
+            RunReadTest(configuration, client, logger, "Read[3]");
+        }
+        finally
+        {
+            DeleteBucket(client, configuration.Bucket);
+        }
+    }
+
+    private static Bucket CreateBucket(string bucketName, StorageClient client)
+    {
+        var bucket = client.CreateBucket(s_projectId, new Bucket { Name = bucketName });
+        SleepAfterBucketCreateDelete();
+        return bucket;
+    }
+
+    private static void DeleteBucket(StorageClient client, string bucketName)
+    {
+        try
+        {
+            client.DeleteBucket(bucketName, new DeleteBucketOptions { UserProject = null, DeleteObjects = true });
+        }
+        catch (GoogleApiException)
+        {
+            // Some tests fail to delete buckets due to object retention locks etc.
+            // They can be cleaned up later.
+        }
+        SleepAfterBucketCreateDelete();
+    }
+
+    private static void CreateRandomFile(Configuration configuration)
+    {
+        var random = new Random();
+        long bytesLeft = configuration.DataSizeInBytes;
+        byte[] buffer = new byte[1024 * 1024];
+        using var output = File.Create(configuration.LocalFile);
+        while (bytesLeft > 0)
+        {
+            random.NextBytes(buffer);
+            int count = (int) Math.Min(bytesLeft, buffer.Length);
+            output.Write(buffer, 0, count);
+            bytesLeft -= count;
+        }
+    }
+
+    private static void RunWriteTest(Configuration configuration, StorageClient client, PerformanceLogger logger, string opName)
+    {
+        string elapsedTimeUs = "-";
+        bool success = false;
+        UploadObjectOptions options = null;
+        if (configuration.UploadChunkSizeInBytes != 0)
+        {
+            options = new UploadObjectOptions { ChunkSize = configuration.UploadChunkSizeInBytes };
+        }
+
+        // Values in the same order as s_logHeaders
+        string[] rowValues =
+        {
+            opName,
+            configuration.DataSizeInBytes.ToString(),
+            "NA", // AppBufferSize,
+            (options?.ChunkSize ?? ResumableUpload.DefaultChunkSize).ToString(), // LibBufferSize
+            "TRUE", // Crc32cEnabled
+            "NA", // MD5Enabled
+            "JSON", // ApiName
+            "-1", // CpuTimeUs,
+            null, // ElapsedTimeUs (populated later)
+            null // Status (populated later)
+        };
+
+        try
+        {
+            CreateRandomFile(configuration);
+            long totalElapsedMicroseconds = 0;
+            Console.WriteLine("Uploading..");
+            using var input = File.OpenRead(configuration.LocalFile);
+            var stopwatch = Stopwatch.StartNew();
+            client.UploadObject(configuration.Bucket, configuration.ObjectName, "application/binary", input, options);
+            totalElapsedMicroseconds = stopwatch.Elapsed.Ticks / TicksPerMicrosecond;
```

Comment:
I don't think we need this local variable at all - or if we want it for clarity, we could at least declare and initialize it at the same place.